client/option.go

```diff
@@ -381,14 +382,36 @@ func WithRetryMethodPolicies(mp map[string]retry.Policy) Option {
 // But if your retry policy is enabled by remote config, WithSpecifiedResultRetry is useful.
 func WithSpecifiedResultRetry(rr *retry.ShouldResultRetry) Option {
 	return Option{F: func(o *client.Options, di *utils.Slice) {
-		if rr == nil {
-			return
+		if rr == nil || (rr.RespRetry == nil && rr.ErrorRetry == nil) {
+			panic(fmt.Errorf("WithSpecifiedResultRetry: invalid '%+v'", rr))
 		}
 		di.Push(fmt.Sprintf("WithSpecifiedResultRetry(%+v)", rr))
 		o.RetryWithResult = rr
 	}}
 }
 
+// WithFallback is used to set the fallback policy for the client.
+// Below is use demo, eg:
+//
+//   demo1. fallback for error and resp
+//	    `client.WithFallback(fallback.NewFallbackPolicy(yourFBFunc))`
+//
+//	 demo2. fallback for error and enable reportAsFallbackRet, reportAsFallbackRet set true the RPC stat will set error as fallback not original error
```

Comment:
set reportAsFallbackRet as true to report the error as fallback error instead of the original error.

Reply:
done